packages/react-ui/src/widgets/RangeWidgetUI.js

```diff
@@ -0,0 +1,189 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Box, makeStyles, Slider, TextField } from '@material-ui/core';
+import { debounce } from '@carto/react-core';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    position: 'relative'
+  },
+  sliderWithThumb: {
+    '& .MuiSlider-rail': {
+      color: theme.palette.text.hint
+    },
+    '& .MuiSlider-track': {
+      display: 'none'
+    }
+  },
+  sliderNotThumb: {
+    pointerEvents: 'none',
+    position: 'absolute',
+    left: 0,
+    right: 0,
+    '& .MuiSlider-thumb': {
+      display: 'none'
+    },
+    '& .MuiSlider-rail': {
+      display: 'none'
+    }
+  },
+  sliderLimits: {
+    zIndex: 1,
+    '& .MuiSlider-track': {
+      color: theme.palette.primary.main,
+      opacity: 0.38
+    }
+  },
+  sliderIntersection: {
+    zIndex: 2
+  },
+  input: {
+    maxWidth: theme.spacing(9),
+    margin: 0,
+    '& fieldset': {
+      borderWidth: 1
+    }
+  }
+}));
+
+function RangeWidgetUI({ values, min, max, limits, onSelectedRangeChange }) {
```

Comment:
I think that in other cases we use `data` instead of values for the UI level (eg HistogramWidgetUI). Maybe we can change, so it’s more homogeneous with others. What do you think?

Reply:
I would appreciate a small conversation together to understand this better tomorrow Javi, before we move on.

For example, why this image has a higher value at left than right?

![](https://user-images.githubusercontent.com/7164554/179529672-96267038-f6d2-4161-8880-c95a20d51b67.png)

Reply:
- I've changed value by data
- I define min & max with 2 different parameters because HistogramWidgetUI does but it's true that we don't support setting min without max, both are mandatories.
- I define limits as an array because limitMin and limitMax sound strange but we can talk about this

```
For example, why this image has a higher value at left than right?
```
The right number is greater than the left number but the right one is cut off because is too big